src/storage/azure.js

```diff
@@ -38,40 +38,35 @@ const downloadFileAsJson = ({ blobServiceClient }) => async (
   fileName
 ) => {
   try {
-    return await downloadFileAsString({blobServiceClient})(bucketName, fileName)
-        .then((data) => JSON.parse(data));
+    return await downloadFileAsString({ blobServiceClient })(
+      bucketName,
+      fileName
+    ).then((data) => JSON.parse(data));
   } catch (e) {
     return Promise.reject("File is not in JSON format");
   }
 };
 
 const downloadFileAsString = ({ blobServiceClient }) => async (
-    bucketName,
-    fileName
+  bucketName,
+  fileName
 ) => {
-    const downloadResponse = await blobServiceClient
-        .getContainerClient(bucketName)
-        .getBlockBlobClient(fileName)
-        .download(0);
-    try {
-        return await streamToString(
-            downloadResponse.readableStreamBody
-        )
-    } catch (e) {
-        return Promise.reject("Could not stringify blob");
-    }
+  const downloadResponse = await blobServiceClient
+    .getContainerClient(bucketName)
+    .getBlockBlobClient(fileName)
+    .download(0);
+  try {
+    return await streamToString(downloadResponse.readableStreamBody);
+  } catch (e) {
+    return Promise.reject("Could not stringify blob");
+  }
 };
 
-const getBlobProperties = ({ blobServiceClient }) => async (
-    bucketName,
-    fileName
-) => {
-    const properties = await blobServiceClient
-        .getContainerClient(bucketName)
-        .getBlockBlobClient(fileName)
-        .getProperties();
-    return properties;
-};
+const getBlobProperties = ({ blobServiceClient }) => (bucketName, fileName) =>
```

Comment:
When you call this function with an await i.e. await getBlobProperties, will it still await a promise resolution? Is that because the .getProperties() returns a promise?

Reply:
yes
there is an eslint rule for it but it doesn't trigger because of the assignment to a variable.

btw the case above is different becasue of the try-catch, so this is a good example when `return await` makes sense (the only example?).

Reply:
I get you! Thanks for the explanation!